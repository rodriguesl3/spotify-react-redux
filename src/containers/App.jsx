import React, { Component } from 'react';

import './App.css';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware, compose } from 'redux';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import { Provider } from 'react-redux'
import rootReducers from '../redux-flow/reducers/index'

import { NavBar } from '../components/navbar'
import Login from './Authenticate'

const store = createStore(rootReducers,
  compose(applyMiddleware(thunk), window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
);


const App = () => {
  return (
    <Provider store={store}>
      <NavBar brandName="N2L" />
      <div className="container">
        <Router>
          <Route path="/login" component={Login} />
          {/* <Route path="/login" component={Login} />
        <Route path="/main" component={Dashboard} /> */}
        </Router>
      </div>

    </Provider>
  );
}
export default App;
