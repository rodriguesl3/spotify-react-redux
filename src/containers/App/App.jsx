/* eslint-disable no-undef */
/* eslint-disable no-underscore-dangle */
import React from 'react';

import './App.css';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware, compose } from 'redux';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import rootReducers from '../../redux-flow/reducers/index';


import Login from '../Authenticate';
import MainContainer from '../MainContainer/MainContainer';


const store = createStore(rootReducers, compose(applyMiddleware(thunk)));

// const store = createStore(rootReducers,
//   compose(applyMiddleware(thunk),
//     window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()));

const App = () => (
  <Provider store={store}>
    <Router>
      <Switch>
        <div>
          <Route exact path="/" component={Login} />
          <Route path="/main" component={MainContainer} />
        </div>
      </Switch>
    </Router>
  </Provider>
);

export default App;
