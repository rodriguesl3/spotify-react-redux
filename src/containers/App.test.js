/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });


test('App snaphot test', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
