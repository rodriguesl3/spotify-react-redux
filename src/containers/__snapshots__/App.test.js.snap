// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snaphot test 1`] = `<div />`;
